src/05-derivatives/01-introduction.md

```diff
@@ -0,0 +1,221 @@
+every derivatives dataset is considered a BIDS dataset and must include a
+`dataset_description.json` file at the root level. Consequently, files should be
+organized to comply with BIDS to the full extent possible (i.e., unless
+explicitly contradicted below). Any subject-specific derivatives should be
+housed within each subject’s directory; if session-specific derivatives are
+generated, they should be deposited under a session subdirectory within the
+corresponding subject directory; and so on.
+
+## Derived dataset and pipeline description
+
+As for any BIDS dataset a `dataset_description.json` file MUST be found at the
+top level of the particular pipeline:
+`<dataset>/derivatives/<pipeline_name>/dataset_description.json`
+
+In addition to raw BIDS datasets, derived BIDS datasets include the following
+required or recommended `dataset_description.json` keys (a dot in the Key name
+denotes a key in a subdictionary):
+
+| **Key name**                                | **Description**                                                                                                                                                                                                                              |
+| ------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| PipelineDescription.Name                    | REQUIRED. Name of the pipeline that generated the outputs. In case the derived dataset is stored as a subfolder of the raw dataset this field MUST be a substring of the derived dataset folder name (a.k.a. `<pipeline_name>` - see above). |
+| PipelineDescription.Version                 | OPTIONAL. Version of the pipeline.                                                                                                                                                                                                           |
+| PipelineDescription.CodeURL                 | OPTIONAL. URL where the code for the analysis can be found.                                                                                                                                                                                  |
+| PipelineDescription.DockerHubContainerTag   | OPTIONAL. Docker Hub tag where the software container image used in this analysis can be found.                                                                                                                                              |
+| PipelineDescription.SingularityContainerURL | OPTIONAL. URL where the Singularity software container image used in this analysis can be found.                                                                                                                                             |
```

Comment:
Is there a strong reasoning behind providing a Docker container reference via tag, but a Singularity container reference via URL? While these may conform to expected current usage, I don't see any logical reason why one could not e.g. provide a URL for a Docker container that's hosted somewhere other than DockerHub, or give a tag for a Singularity container hosted somewhere like Singularity Hub. Would having { `PipelineDescription.ContainerTag`, `PipelineDescription.ContainerURL` } not serve the same purpose but be less restrictive / prescriptive?

Reply:
@Lestropie I've updated with `PipelineDescription.Container` that mirrors the form of `SourceDatasets`. I can flesh out the subfields, but if we do it for one, we should probably do it for both.